includes/html/api_functions.inc.php

```diff
@@ -1161,11 +1161,40 @@ function list_alert_rules(Illuminate\Http\Request $request)
     $sql = '';
     $param = [];
     if ($id > 0) {
-        $sql = 'WHERE id=?';
+        $sql = 'WHERE `a`.`id`=?';
         $param = [$id];
     }
 
-    $rules = dbFetchRows("SELECT * FROM `alert_rules` $sql", $param);
```

Comment:
Eloquent goes something like this (but for all relationships), but you need #14500
```php
$rules = AlertRule::with(['devices' => function ($query) {
    return $query->select('devices.device_id');
}])->get();
```

Reply:
OK thanks for your answer. I tried this and in the API I have some strange results : 
For a request with 2 devices and a group I get
`"devices": [
				{
					"device_id": 1,
					"pivot": {
						"rule_id": 25,
						"device_id": 1
					}
				},
				{
					"device_id": 2,
					"pivot": {
						"rule_id": 25,
						"device_id": 2
					}
				}
			]`
It misses the groups. I assume that it is normal as we did not put the group in the Eloquent query.

But worse, instead of only returning the device ids, it returns these pivot thing..

A working answer from my old SQL way is : 
`"devices": [
				1,
				2
			],
			"groups": [
				6
			]`

And if there is all devices selected it return an empty array as devices. It may break something if we reinject this Json code into a PUT request.

Reply:
Sorry, I'll give you more complete code.  I only gave you a bit to get you started.

Reply:
To make this as clean as possible, I had to use a json resource:
`php artisan make:resource AlertRule`
```php
    public function toArray($request)
    {
        $rule = parent::toArray($request);
        $rule['devices'] = $this->devices->pluck('device_id')->all();
        $rule['groups'] = $this->groups->pluck('id')->all();
        $rule['locations'] = $this->locations->pluck('id')->all();

        return $rule;
    }
```

Then, the api method becomes just:
```php
function list_alert_rules(Illuminate\Http\Request $request)
{
    $id = $request->route('id');

    $rules = \App\Http\Resources\AlertRule::collection(
        \App\Models\AlertRule::when($id, fn($query, $id) => $query->where('id', $id))
        ->with(['devices:device_id', 'groups:id', 'locations:id'])->get()
    );

    return api_success($rules->toArray($request), 'rules');
}
```